File: data/code_pairs/cpp/2505.cpp
#include <iostream>
#include <complex>
#include <cstdlib>

int main() {
    std::complex<double> z0, z1, z2;

    z0 = std::complex<double>(0.0, 0.5);
    z1 = 1.0 / z0;
    if (z1 != std::complex<double>(0.0, -2.0)) {
        std::cerr << "STOP 1" << std::endl;
        std::exit(1);
    }

    z0 = 10.0 * z0;
    if (z0 != std::complex<double>(0.0, 5.0)) {
        std::cerr << "STOP 2" << std::endl;
        std::exit(2);
    }

    z2 = std::complex<double>(1.0, 2.0);
    z1 = z0 / z2;
    if (z1 != std::complex<double>(2.0, 1.0)) {
        std::cerr << "STOP 3" << std::endl;
        std::exit(3);
    }

    z1 = z0 * z2;
    if (z1 != std::complex<double>(-10.0, 5.0)) {
        std::cerr << "STOP 4" << std::endl;
        std::exit(4);
    }

    return 0;
}

File: data/code_pairs/fortran/2505.f90
program foo
    implicit none
    complex(kind=8) z0, z1, z2

    z0 = (0.0d0, 0.5d0)
    z1 = 1.0d0 / z0
    if (z1 .ne. (0.0d0, -2.0d0)) stop 1

    z0 = 10.0d0 * z0
    if (z0 .ne. (0.0d0, 5.0d0)) stop 2

    z2 = (1.0d0, 2.0d0)
    z1 = z0 / z2
    if (z1 .ne. (2.0d0, 1.0d0)) stop 3

    z1 = z0 * z2
    if (z1 .ne. (-10.0d0, 5.0d0)) stop 4

end program foo